wiki/markdown/usage/service.md

```diff
@@ -204,9 +204,12 @@ https://localhost:8080/eventrepository/events/?meta.type=EiffelArtifactCreatedEv
                 }]
 ```
 
-#### Lookups are provided with two options:
+#### Lookups are provided with four options:
 
 | Options              | Default Value | Description                                                                                                                                                                                                                                                                                                                                                                                                                            |
 |----------------------|---------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | failIfMultipleFound: | False         | If value is set to false, REMReM generates event if multiple event ids found or no event id fetched from the event repository configured in generate and adds them in links section of the target field. If value is set to true, if only one event is found then REMReM generates the event successfully and if multiple event ids are found then REMReM fails to generate any event and displays relevant error message to the user. |
-| failIfNoneFound:     | False         | If value is set to true, no event ids fetched from event repository , REMReM fails to generate any event and displays relevant error message to the user. If value is set to false,REMReM generates event if no event found or atleast one event fetched from the event repository configured in configuration file.                                                                                                                   |
\ No newline at end of file
+| failIfNoneFound:     | False         | If value is set to true, no event ids fetched from event repository , REMReM fails to generate any event and displays relevant error message to the user. If value is set to false,REMReM generates event if no event found or atleast one event fetched from the event repository configured in configuration file.|                                                                                                 
+| shallow:             | True          | This parameter is to control the behavior of ER towards external ERs,default is true which states the ER queries can be connected to internal ER.else the ER queries can be connected to external ERs.|
```

Comment:
It is not clear what an external ER and internal ER is in this case. Please clarify this comment.

Reply:
Internal ER means which is specifically configured in REMReM generate 
External ER means  which are configured as external ER's to above specifically connected REMReM ER

Reply:
The parameter should be renamed to 'connectToExternalERs' with default value False. The description could be:

> If value is set to True then REMReM will query external ERs and not just the locally used ER. The reason for the default value to be False is to decrease the load on external ERs. <add a reference to documentation about external/local ERs here as well>

Reply:
> The parameter should be renamed to 'connectToExternalERs' with default value False. The description could be:
> 
> > If value is set to True then REMReM will query external ERs and not just the locally used ER. The reason for the default value to be False is to decrease the load on external ERs. <add a reference to documentation about external/local ERs here as well>

Nice, I like this suggestion!

Reply:
Done, but with little  change connectToExternalER's default value is set to True it never fetch from external ER's

Reply:
That doesn't sound intuitive. If 'connectToExternalERs' is True it should connect to external ERs (by setting shallow=false to ER). If it is false it should not (by setting shallow=True to ER)

Reply:
> That doesn't sound intuitive. If 'connectToExternalERs' is True it should connect to external ERs (by setting shallow=false to ER). If it is false it should not (by setting shallow=True to ER)

Done and pushed the changes